packages/router/utils.ts

```diff
@@ -374,36 +374,53 @@ function rankRouteBranches(branches: RouteBranch[]): void {
 }
 
 const paramRe = /^:\w+$/;
-const dynamicSegmentValue = 3;
-const indexRouteValue = 2;
+const partialParamRe = /:\w+$/;
+const splatRe = /^\*$/;
+const partialSplatRe = /\*$/;
+
+const staticSegmentValue = 10; /*         /static         */
+const partialDynamicSegmentValue = 9; /*  /prefix-:param  */
+const partialSplatValue = 8; /*           /prefix-*       */
+const dynamicSegmentValue = 3; /*         /:param         */
+const indexRouteValue = 2; /*             /               */
 const emptySegmentValue = 1;
-const staticSegmentValue = 10;
-const splatPenalty = -2;
-const isSplat = (s: string) => s === "*";
+const splatPenalty = -2; /*               /*              */
+const partialSplatPenalty = -1; /*        /prefix-*       */
 
 function computeScore(path: string, index: boolean | undefined): number {
   let segments = path.split("/");
-  let initialScore = segments.length;
-  if (segments.some(isSplat)) {
```

Comment:
splat scoring is handled inside the below loop to avoid multiple traversals of `segments`